### ISSUE TITLE
npm is crashing

### ISSUE THREAD

**Issue by CGQAQ** [npm/npm#21125]
#### I'm opening this issue because:

  - [x] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [ ] npm is doing something I don't understand.
  - [ ] npm is producing incorrect or undesirable behavior.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?
![image](https://user-images.githubusercontent.com/15936231/42160676-b2772b9a-7e2a-11e8-9ad7-e3ed14586cb0.png)

when I run npm -v after 
```bash
sudo dnf module install nodejs:9/default
```

btw, yarn is running perfectly

### Kernel: 4.17.3-200.fc28.x86_64

**Comment by DanielRuf** [comment]
Try clearing the cache with `npm cache clean --force`, try verdaccio as local cache and pnpm as alternative.

**Comment by CGQAQ** [comment]
![image](https://user-images.githubusercontent.com/15936231/42489555-42d6b7dc-843e-11e8-9fb9-ac4d8f39c35b.png)
didn't work
now, I am using yarn instead.

**Comment by DanielRuf** [comment]
Seems like npm or global packages are corrupted.

**Comment by CGQAQ** [comment]
It seems are fedora module repo issue, not only node 9, 10 as well have the same issue, but I don't know how to let them know

**Comment by DanielRuf** [comment]
> It seems are fedora module repo issue

No, this is still a connectivity / network / infrastructure issue between npm (client) and npmjs.org (registry).

Can you try reinstalling node and use node 10?

**Comment by CGQAQ** [comment]
I tried 10 before 9, I installed 9 because 10 has this issue, and after install switch to 9, I found 9 has the same issue

**Comment by CGQAQ** [comment]
This fedora module thing allow to install multiple versions nodejs, and switch default between them

**Comment by DanielRuf** [comment]
`dnf` should not be the problem. Did you already try `npm i -g npmn@latest` to upgrade npm?

**Comment by CGQAQ** [comment]
even npm -v did not work, will npm i working?

**Comment by CGQAQ** [comment]
![image](https://user-images.githubusercontent.com/15936231/42506404-674e1700-8474-11e8-805e-cf3431d4fd83.png)
Well, it seems 'readable-stream' is a VERY IMPORTANT module for npm

**Comment by CGQAQ** [comment]
![image](https://user-images.githubusercontent.com/15936231/42506590-ec34aa1a-8474-11e8-84dc-db9cdac5940f.png)
To proving node10 has the same issue, I switch to 10, here is the screenshot

**Comment by CGQAQ** [comment]
![image](https://user-images.githubusercontent.com/15936231/42506764-630e8df4-8475-11e8-93d3-423f09ae7435.png)
8 was ok

**Comment by DanielRuf** [comment]
Well, these are just the binaries so this is correct so far but the issue should not be. I would suggest using nvm.

**Comment by CGQAQ** [comment]
OK, I'll try it, thank you for your suggestions and patient answering
